Geo support

SebastianStehle opened the pull request.

SebastianStehle commented:
I have not worked on  sqlite yet, but it should look very similar. The problem is to install the binaries here.

PhenX commented:
I seems like adding 

    <PackageReference Include="SQLitePCLRaw.bundle_e_sqlite3" Version="2.1.11" />

To the `Tests` package makes it work also for Sqlite

SebastianStehle commented:
I don't know what is wrong now. I am really lost. I can't find the damn error :(

SebastianStehle commented:
Okay, I am done. 

I have also simplified the test fixtures a little bit and fixed a bug in the metadata provider with multiple contexts. I still have not figured out how to make Sqlite workign locally and in github, but this is probably something for another PR.

PhenX commented:
OK I'll review it tonight

PhenX commented:
I'm not a big fan of the NetTopologySuite dependency, did you consider resolving a ITypeMappingSource? The one for Sql server for example https://github.com/dotnet/efcore/blob/main/src/EFCore.SqlServer/Storage/Internal/SqlServerTypeMappingSource.cs

SebastianStehle commented:
I don#t get it. How would you make the conversion then?

PhenX commented:
Well EF knows how to convert geo types when we call UseNetTopologySuite, so I think we should use these converters

SebastianStehle commented:
How?

SebastianStehle commented:
For exmaple we already use this method to get the correct type for postgres:

```

        var mapping = column.Property.GetRelationalTypeMapping() as NpgsqlTypeMapping;

        return mapping?.NpgsqlDbType;
        ```
        
        But for Geography we get no mapping at all.

PhenX commented:
At least for NPGSQL, I'll have to check for others, but we can use the mapping to get the underlying NpgsqlDbType, which seems to be different for bulk insert (bytea, seeing your code)

![image](https://github.com/user-attachments/assets/bb93d74c-689b-483d-9fef-ce07f78c3ac9)

PhenX commented:
Let me go further, I'm not sure I went deep enough in your PR

SebastianStehle commented:
Sure, I basically went from error to error. Always got something like "Cannot convert from x to y"

PhenX commented:
Ok, I thought I could find something among mapping sources, but no...

SebastianStehle commented:
There are internal APIs, they are called ProviderConverter, but I could not find any way to get access to them.

SebastianStehle commented:
A release would be awesome.
